tests/share/test_secretsmanager.py

```diff
@@ -0,0 +1,239 @@
+                            "Code": "ResourceNotFoundException",
+                        }
+                    },
+                    "GetSecretValue",
+                )
+
+            if secrets["type"] == "SecretBinary":
+                secret_data = base64.b64decode(secrets["data"]).decode("utf-8")
+            elif secrets["type"] == "SecretString":
+                secret_data = secrets["data"]
+
+        return secret_data
+
+
+class TestAWSSecretsManager(TestCase):
+    @mock.patch(
+        "share.secretsmanager.get_secret_values",
+        new=MockContent.get_secret_values,
+    )
+    def test_parse_secrets_manager(self) -> None:
+        with self.subTest("invalid arn format - too long"):
+            config_yaml = """
+                inputs:
+                    - type: sqs
+                    id: "arn:aws:secretsmanager:eu-central-1:123-456-789:secret:sqs_secret:THIS:IS:INVALID"
+                    outputs:
+                        - type: elasticsearch
+                        args:
+                            elasticsearch_url: "arn:aws:secretsmanager:eu-central-1:123-456-789:secret:
+                            es_secrets:elasticsearch_url"
+                            username: "arn:aws:secretsmanager:eu-central-1:123-456-789:secret:
+                            es_secrets:elasticsearch_url"
+                            password: "arn:aws:secretsmanager:eu-central-1:123-456-789:secret:es_secrets:password"
+                            dataset: "dataset"
+                            namespace: "namespace"
+            """
+
+            with self.assertRaisesRegex(
+                SyntaxError,
+                "Invalid arn format: arn:aws:secretsmanager:eu-central-1:123-456-789:"
+                + "secret:sqs_secret:THIS:IS:INVALID",
+            ):
+                aws_sm_expander(config_yaml)
+
+        with self.subTest("invalid arn format - pattern not caught"):
+            config_yaml = """
+                inputs:
+                - type: sqs
+                    id: "aws:arn:secretsmanager:eu-central-1:123-456-789:secret:plain_secret"
+                    outputs:
+                    - type: elasticsearch
+                        args:
+                            elasticsearch_url: "arn:aws:secrets_manager:eu-central-1:123-456-789:secret:es_secrets:url"
+                            username: "arn:aws:secretsmanager:eu-central-1:123-456-789:secret:es_secrets:username"
+                            password: "arn:aws:secretsmanager:eu-central-1:123-456-789:secret:es_secrets:password"
+                            dataset: "dataset"
+                            namespace: "namespace"
+            """
+            mock_fetched_data = aws_sm_expander(config_yaml)
+
+            parsed_config_yaml = """
+                inputs:
+                - type: sqs
+                    id: "aws:arn:secretsmanager:eu-central-1:123-456-789:secret:plain_secret"
```

Comment:
`aws:arn` instead of `arn:aws` is a little hard to spot out immediately :)

please add a comment about why the pattern does not match
like `#BEWARE aws:arn != arn:aws` :)

Reply:
is the pattern case sensitive?

Reply:
yes, it is. It will not match the pattern if you enter `Arn:aws...`